src/engraving/rendering/score/systemlayout.cpp

```diff
@@ -1788,21 +1774,14 @@ void SystemLayout::updateCrossBeams(System* system, LayoutContext& ctx)
 
 void SystemLayout::restoreTiesAndBends(System* system, LayoutContext& ctx)
 {
-    std::vector<Segment*> segList;
+    ElementsToLayout elements(system);
     for (MeasureBase* mb : system->measures()) {
-        if (!mb->isMeasure()) {
-            continue;
-        }
-        for (Segment& seg : toMeasure(mb)->segments()) {
-            if (seg.isChordRestType()) {
-                segList.push_back(&seg);
-            }
+        if (mb->isMeasure()) {
+            collectElementsToLayout(toMeasure(mb), elements, ctx);
```

Comment:
Collecting _all_ elements in a system when we know we only want the system, measures, and chords seems a waste of time on the surface. I doubt the performace/space improvements gained with only collecting what we need here would amount to much.  Maybe something to bear in mind when you want to squeeze some more performance out of system layout!